### PR TITLE
Add equality assert syntax

### PR DESCRIPTION
**Pull request by rvanvelzen**
As supported in Psalm: https://psalm.dev/docs/annotating_code/assertion_syntax/#equality-assertions

### CODE REVIEW

File: src/Ast/PhpDoc/AssertTagMethodValueNode.php [hunk 1]
```diff
@@ -23,23 +23,28 @@ class AssertTagMethodValueNode implements PhpDocTagValueNode
...
 		$this->description = $description;
 	}
 
 
 	public function __toString(): string
 	{
 		$isNegated = $this->isNegated ? '!' : '';
-		return trim("{$isNegated}{$this->type} {$this->parameter}->{$this->method}() {$this->description}");
+		$isEquality = $this->isEquality ? '=' : '';
+		return trim("{$isNegated}{$isEquality}{$this->type} {$this->parameter}->{$this->method}() {$this->description}");
```
Comment (staabm) [inline src/Ast/PhpDoc/AssertTagMethodValueNode.php:47]: seeing this, I am wondering whether paranthesis as I used them in

https://github.com/phpstan/phpstan-src/pull/1880/files#diff-8f0c25b196228aa9450cbd3c1d8c83d6d5f33c9fe0aa1b07effe32d41275dd3eR41

are a valid thing?

Comment (rvanvelzen) [reply src/Ast/PhpDoc/AssertTagMethodValueNode.php:47]: They're valid since you can just parenthesize any type, but it's not necessary.

Comment (staabm) [reply src/Ast/PhpDoc/AssertTagMethodValueNode.php:47]: additionally, maybe its worth having a test-case for more complex ones like e.g. `@phpstan-assert-if-false !=(float|int|numeric-string) $value`

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**